runtime/util/hshelp.c

```diff
@@ -1726,17 +1722,48 @@ fixRAMConstantPoolForFastHCR(J9ConstantPool *ramConstantPool, J9HashTable *class
 									methodResult = hashTableFind(methodHashTable, &methodPair);
 									if (NULL != methodResult) {
 										UDATA argCount = (methodRef->methodIndexAndArgCount & 255);
-										UDATA newMethodIndex = getITableIndexForMethod(methodResult->newMethod, resolvedClass);
+										J9Method *newMethod = methodResult->newMethod;
+										UDATA newMethodIndex = newMethod - J9_CLASS_FROM_METHOD(newMethod)->ramMethods;
 										/* Fix the index in the resolved CP entry, retaining the argCount */
 										methodRef->methodIndexAndArgCount = ((newMethodIndex << J9_ITABLE_INDEX_SHIFT) | argCount);
```

Comment:
Does this need the `J9_ITABLE_INDEX_METHOD_INDEX` or'd into the value ?

Reply:
Yes it does.